### ISSUE TITLE
Wrong visualisation of point cloud after calibration

### ISSUE THREAD

**Issue by TongZhangTHU** [IFL-CAMP/easy_handeye#127]
Hello,

After I calibrated the RealSense camera extrinsics relative to the Franka Emika Panda using easy_handeye, I visualized the point cloud in the robot frame with rviz /camera/depth/color/points topic (I have published the extrinsics TF), but found that the point cloud is in the wrong position and looks rotated in some way, I suspect that it is caused by the wrong coordinate system. I wonder if you have any suggestions for my problem?

![img_v2_93d4968f-2daf-440e-bd2d-633891baf4ag](https://github.com/IFL-CAMP/easy_handeye/assets/49636953/2ac6dcb6-155a-468d-9d39-bf3f9a3bd0ad)

**Comment by marcoesposito1988** [comment]
Hi @TongZhangTHU,

did you visualize the point cloud after starting the system including `publish.launch`, or is still using the dummy position of the camera from the `calibrate.launch` file?

**Comment by TongZhangTHU** [comment]
@marcoesposito1988 

I visualize the point cloud from the  `/camera/depth/color/points` topic after starting `roslaunch realsense2_camera rs_camera.launch filters:=pointcloud`.

**Comment by marcoesposito1988** [comment]
Yes, but after doing the calibration did you restart everything, publishing the hand-eye calibration with [this launch file](https://github.com/IFL-CAMP/easy_handeye/blob/master/easy_handeye/launch/publish.launch), or is the system still running with `calibrate.launch` instead?

After calibration the new transform is not applied automatically, you have to restart everything using a different launch file to use the result of the calibration.

**Comment by TongZhangTHU** [comment]
@marcoesposito1988 

The results I presented above were obtained after restarting everything and publishing the hand-eye calibration with [this launch file](https://github.com/IFL-CAMP/easy_handeye/blob/master/easy_handeye/launch/publish.launch), just as you suggested.

**Comment by marcoesposito1988** [comment]
It seems that the point cloud is rotated 90 degrees along two axes, so probably a different frame is being used to publish the calibration than what was used during the calibration. Please double check what is passed as frame for the camera

**Comment by Skeptic9793** [comment]
> It seems that the point cloud is rotated 90 degrees along two axes, so probably a different frame is being used to publish the calibration than what was used during the calibration. Please double check what is passed as frame for the camera
I'm also experiencing this problem, which seems to be caused by the fact that realsense's optical coordinate system definition is different from the ROS coordinate system definition

**Comment by Skeptic9793** [comment]
I'm also experiencing this problem, which seems to be caused by the fact that realsense's optical coordinate system definition is different from the ROS coordinate system definition

**Comment by Panda12580** [comment]
> I'm also experiencing this problem, which seems to be caused by the fact that realsense's optical coordinate system definition is different from the ROS coordinate system definition

Have you solved this problem? I ran into this problem too, where I tried to rotate by publishing an intermediate coordinate system and it seemed to work, but it seemed like there was some error in the rotation Angle. Can you tell me your solution?

**Comment by marcoesposito1988** [comment]
You may have to use a different frame for the camera optical center and the reference frame of the marker tracking. Please look here for a reference: https://github.com/IFL-CAMP/easy_handeye/blob/master/docs/example_launch/ur5_kinect_calibration.launch#L21
